openedx/core/djangoapps/safe_sessions/middleware.py

```diff
@@ -368,6 +368,8 @@ def _verify_user(request, userid_in_session):
         given userid_in_session.
         """
         if hasattr(request, 'safe_cookie_verified_user_id'):
+            if hasattr(request.user, 'real_user'):  # User is a masqueraded user.
+                request.user = request.user.real_user
```

Comment:
```suggestion
            if hasattr(request.user, 'real_user'):
                # If a view overrode the request.user with a masqueraded user, this will
                #   revert/clean-up that change during response processing.
                request.user = request.user.real_user
```